### PR TITLE
Feature/agenda content

### PR DESCRIPTION
**Pull request by ArtemisSoftware**

### CODE REVIEW

**Comment by ArtemisSoftware** [comment]
Captura de ecrã 2023-03-16 160415

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaState.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.artemissoftware.tasky.agenda.domain.models.AgendaItem
+import com.artemissoftware.tasky.agenda.domain.models.DayOfWeek
+import java.time.LocalDate
+
+data class AgendaState(
+    val isLoading: Boolean = false,
+    val agendaItems: List<AgendaItem> = emptyList(),
+    val daysOfTheWeek: List<DayOfWeek> = emptyList(),
+    val selectedDayOfTheWeek: LocalDate = LocalDate.now(),
+    val userName: String = ""
```
Comment (philipplackner) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaState.kt:13]: Is the username really part of the state? In the end state is a value that can change over time, but the username is constant from this point on, isn't it? Or is there anything asynchronous involved loading it?

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/mappers/AgendaMapper.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import com.artemissoftware.tasky.agenda.AgendaItemType
+import com.artemissoftware.tasky.agenda.domain.models.AgendaItem
+
+fun AgendaItem.toAgendaItemType(): AgendaItemType{
+    return when(this){
+        is AgendaItem.Reminder -> {
+            AgendaItemType.Reminder()
+        }
+    }
+}
```
Comment (philipplackner) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/mappers/AgendaMapper.kt:12]: Curious what that is needed for 🤔 
Can't you already check what type a given item is using the `AgendaItem` sealed class? 
E.g. `if(agendaItem is AgendaItem.Reminder) { println("I'm a reminder") }`

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaState.kt (hunk 1 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaState.kt:13]: The name I will get from data store. It's load will be asynchronous

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/mappers/AgendaMapper.kt (hunk 2 above)
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/mappers/AgendaMapper.kt:12]: Will refactor

**Pull request merged**